xarray/convert.py

```diff
@@ -157,18 +161,22 @@ def _iris_cell_methods_to_str(cell_methods_obj):
     return ' '.join(cell_methods)
 
 
+def _iris_name(iris_obj):
+    return iris_obj.standard_name or iris_obj.var_name or 'unknown'
```

Comment:
Best using ``cube.name()``, which has almost the same logic.

Reply:
`cube.name()` evaluates as `standard_name or long_name or var_name or STASH attribute or 'unknown'`
`long_name` can be an arbitrary string. I thought it would be a bad idea to use this as name attribute in xarray, but I do not know what the consequences would be, for example when writing the `DataArray` to a netcdf file. I would rather use `'unknown'`, than a very long `long_name` containing whitespace, linebreaks, etc.
Iris has some logic to create a valid netcdf `var_name` when writing to a netcdf file.

Reply:
I'm a little nervous about using `'unknown'` names:
- For `xarray.DataArray` objects, if the name is `'unknown'` we should probably stick with `name=None`, which is xarray's version of unknown names.
- For coordinates, we could use `'unknown'`, but only for at most one coordinate, because names in xarray's data model need to be unique.